ID: 14574
Title: Erroneous Binder Error: `Referenced table ... not found!` when identifier is in scope of formal function argument
Description:
### What happens?

DuckDB handles the following function properly:
```
CREATE FUNCTION complex_div(a, b) AS (
  {re : (a.re * b.re + a.im * b.im) / complex_mag_squared(b),
   im : (a.im * b.re - a.re * b.im) / complex_mag_squared(b) }
);
```
One would have hoped that it would be trivial to avoid the
redundant computation, e.g. via a WITH clause:

```
CREATE OR REPLACE FUNCTION complex_div(a, b) AS (
  WITH denom AS (select complex_mag_squared(b) as denom)
  SELECT {re : (a.re * b.re + a.im * b.im) / denom,
          im : (a.im * b.re - a.re * b.im) / denom }
  FROM denom
);          
```
But DuckDB gets confused:

```
D select complex_div( {re:2.0,im:1.0}, {re:2.0,im:0.0});
Binder Error: Referenced table "a" not found!
```

Please note: I'm aware of several workarounds, so this
is just a bug report.




### To Reproduce

```
CREATE TYPE COMPLEX AS STRUCT(re REAL, im REAL);

CREATE OR REPLACE FUNCTION complex_mag_squared(x) AS (
   x.re * x.re + x.im * x.im
);

CREATE OR REPLACE FUNCTION complex_div(a, b) AS (
  WITH denom AS (select complex_mag_squared(b) as denom)
  SELECT {re : (a.re * b.re + a.im * b.im) / denom,
          im : (a.im * b.re - a.re * b.im) / denom }
  FROM denom
);          

select complex_div( {re:2.0,im:1.0}, {re:2.0,im:0.0});
```

### OS:

macos

### DuckDB Version:

v1.1.3-dev38 et al.

### DuckDB Client:

CLI

### Hardware:

_No response_

### Full Name:

Peter Koppstein

### Affiliation:

Princeton University

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a source build

### Did you include all relevant data sets for reproducing the issue?

Not applicable - the reproduction does not require a data set

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have